leastsquare.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+
+def calculate_weight(basis_function, output_labels):
+
+    # For MW = S
+    S = np.dot(basis_function, output_labels)
+    M = np.dot(basis_function, np.transpose(basis_function))
+
+    # Solve linear equation
+    return np.linalg.solve(M, S)
+
+
+def calculate_error(basis_function, output_labels, weight):
+    predicted_output = np.dot(np.transpose(weight), basis_function)
```

Comment:
Another way to do this is as below, not sure which is more pythonic:
```python
predicted_output = weigth.T@basis_function
```

Reply:
Simplified implementation. Implemented